tdct/deconvolution.py

```diff
@@ -0,0 +1,864 @@
+import sys
+import os
+
+import tifffile
+if getattr(sys, 'frozen', False):
+    # programm runs in a bundle (pyinstaller)
+    execdir = sys._MEIPASS
+else:
+    execdir = os.path.dirname(os.path.realpath(__file__))
+sys.path.append(execdir)
+
+import numpy as np
+from PyQt5 import QtWidgets
+from . import clrmsg
+from . import TDCT_debug
+
+try:
+    import tifffile as tf
+except:
+    sys.exit("Please install tifffile, e.g.: pip install tifffile")
+
+debug = TDCT_debug.debug
+
+import scipy.signal
+import scipy.fft
+
+#Internal helper class to handle progress bar
+class _progrBarHandle():
+    def __init__(self, qtprocessbar, niter):
+        self.iter=0
+        self.niter = niter
+        self.qtbar = qtprocessbar
+
+        # if self.qtbar:
+        # 	self.qtbar.value= self.iter/self.niter*100
+        # 	QtWidgets.QApplication.processEvents()
+        self.updateUI() #Not working
+    def increment(self):
+        self.iter+=1
+        self.updateUI()
+    def updateUI(self):
+        if self.qtbar:
+            self.qtbar.setValue(self.iter/self.niter*100)
+            QtWidgets.QApplication.processEvents()
+    def setmax(self):
+        self.iter= self.niter
+        self.updateUI()
+
+# def _convertAndNormalise(d0):
+#     d1= d0.astype('float32')
+#     vmax= d0.max()
+#     vmin=d0.min()
+#     d2 = (d1-vmin)/(vmax-vmin)
+#     return d2
+
+def _convertAndNormalise(d0, doNormalise=True):
+    d1= d0.astype('float32')
+    ret=d1
+    if doNormalise:
+        vmax= d0.max()
+        vmin=d0.min()
+        d2 = (d1-vmin)/(vmax-vmin)
+        ret=d2
+    return ret
+
+def _centered(arr, newshape):
```

Comment:
Why do you prefer to have functions beginning with "_"? If you don't want these to be exported from the file you can add __all__ = [ ... ] at the top of the file to explicitly state which functions and classes to export. https://stackoverflow.com/questions/44834/can-someone-explain-all-in-python